src/views/tutorial-view/index.tsx

```diff
@@ -74,48 +77,60 @@ export default function TutorialView({
     video,
     collectionCtx,
   } = tutorial
+
+  const InteractiveLabWrapper = handsOnLab?.id ? InstruqtProvider : Fragment
+
   return (
-    <SidebarSidecarLayout
-      breadcrumbLinks={layout.breadcrumbLinks}
-      sidebarSlot={
-        <Sidebar
-          title={collectionCtx.current.shortName}
-          menuItems={collectionCtx.current.tutorials.map((t) =>
-            formatTutorialToMenuItem(t, collectionCtx.current.slug, currentPath)
-          )}
-        />
-      }
-      headings={layout.headings}
+    <InteractiveLabWrapper
+      key={slug}
+      {...(handsOnLab?.id && { labId: handsOnLab?.id })}
     >
-      <header id="overview">
-        <Heading
-          level={1}
-          size={500}
-          weight="bold"
-          slug={layout.headings[0].slug}
-          id={slug}
-        >
-          {name}
-        </Heading>
-        <Badges
-          tutorialMeta={{
-            readTime,
-            products: productsUsed.map((p) => p.product.slug),
-            isBeta: getIsBeta(productsUsed),
-            edition,
-            hasVideo: Boolean(video),
-            isInteractive: Boolean(handsOnLab),
-          }}
+      <SidebarSidecarLayout
+        breadcrumbLinks={layout.breadcrumbLinks}
+        sidebarSlot={
+          <Sidebar
+            title={collectionCtx.current.shortName}
+            menuItems={collectionCtx.current.tutorials.map((t) =>
+              formatTutorialToMenuItem(
+                t,
+                collectionCtx.current.slug,
+                currentPath
+              )
+            )}
+          />
+        }
+        headings={layout.headings}
+      >
+        <header id="overview">
+          <Heading
+            level={1}
+            size={500}
+            weight="bold"
+            slug={layout.headings[0].slug}
+            id={slug}
+          >
+            {name}
+          </Heading>
+          <Badges
+            tutorialMeta={{
+              readTime,
+              products: productsUsed.map((p) => p.product.slug),
+              isBeta: getIsBeta(productsUsed),
+              edition,
+              hasVideo: Boolean(video),
+              isInteractive: Boolean(handsOnLab),
+            }}
+          />
+          <InteractiveLabToggle showButton={Boolean(handsOnLab?.id)} />
```

Comment:
Thoughts on removing the `showButton` prop and conditionally rendering this component instead?

```tsx
{isInteractive && <InteractiveLabToggle />}
```

Also seeing that the `isInteractive` logic is defined in two different ways, we probably want just one:

- `Boolean(handsOnLab)` (in the `tutorialMeta` object just above this)
- `Boolean(handsOnLab?.id)`